### PR TITLE
add id attr settings extensions

### LINKED ISSUE #19869: Add `id` attributes to setting sections in `/settings/extensions` for anchor links
## Is this feature related to a problem? Please describe.

- When we're linking users to certain settings, we should be able to append an anchor link e.g. `/settings/extensions#api` to present the user with the precise section we need.
- We should ideally have this for all `/settings` but can start here for Pro Tools onboarding purposes

## Describe the solution you’d like

Perhaps, something like the following:

- `#templates` or `#comment-templates` for "Comment templates"
- `#rss` for "Publishing to DEV Community from RSS"
- `monetization` for "Web monetization"
- `api` for "Dev community API keys".

## Definition of Done

- [ ] Add `id` attributes to each section to support anchor links to the specific setting in `/settings/extensions`

## Additional Context

- This is required to make it easy to extract an API key to aid in Pro Tools onboarding

### PR DESCRIPTION
**Pull request by narender2031**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Add id attributes to setting sections in /settings/extensions

## Related Tickets & Documents

- Closes #19869 

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, a note
on the devices and browsers this has been tested on, as well as any relevant
images for UI changes._

### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_
- [ ] Semantic HTML implemented?
- [ ] Keyboard operability supported?
- [ ] Checked with [axe DevTools](https://www.deque.com/axe/) and addressed `Critical` and `Serious` issues?
- [ ] Color contrast tested?

_For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

- [ ] Yes
- [x] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

### CODE REVIEW

File: app/views/users/_integrations_github_repositories.html.erb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <% if show_integration %>
-  <div class="crayons-card crayons-card--content-rows">
+  <div class="crayons-card crayons-card--content-rows" id="github-repos">
```
Comment (rt4914) [inline app/views/users/_integrations_github_repositories.html.erb:2]: Change id from `github-repos` to `github` for simplicity.

File: app/views/users/_response_templates.html.erb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<div class="crayons-card crayons-card--content-rows">
+<div class="crayons-card crayons-card--content-rows" id="#comment-templates">
```
Comment (rt4914) [inline app/views/users/_response_templates.html.erb:1]: Remove `#` from the id name. This was written with some different context in the original issue.

File: app/views/users/_web_monetization.html.erb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <% if policy(Article).create? %>
-  <div class="crayons-card crayons-card--content-rows">
+  <div class="crayons-card crayons-card--content-rows" id="web_monetization">
```
Comment (rt4914) [inline app/views/users/_web_monetization.html.erb:2]: Change id to `monetization` for simplicity.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@narender2031 Thanks a lot.
I have suggested minor changes for this PR. Do let me know if you need any help or clarification. Also really happy to see that your second contribution this month. 🚀

File: app/views/users/_response_templates.html.erb (hunk 2 above)
Comment (narender2031) [reply app/views/users/_response_templates.html.erb:1]: oops! I missed this one during copy paste.

File: app/views/users/_api_keys.html.erb [hunk 4]
```diff
@@ -1,22 +1,22 @@
-<div class="crayons-card crayons-card--content-rows">
+<div class="crayons-card crayons-card--content-rows" id="api-key>
```
[the remark this answers is not in the document]
Comment (narender2031) [reply app/views/users/_api_keys.html.erb]: @rt4914 The other change in the file has been introduced by the pre-commit hook.

**Review by rt4914 — APPROVED** [review]
Thanks a lot @narender2031 .
Looks good to me :-)

**Review by benhalpern — APPROVED** [review]

**Pull request merged**